rfc/hierarchical-capabilities.md

```diff
@@ -0,0 +1,57 @@
+# Hierarchical Capabilities
+
+## Abstract
+
+Currently UCAN capabilities described in [W3Up specification](https://github.com/web3-storage/specs) do not follow consistent organizational structure. To make things worse accounts on login delegate `*` capabilities and spaces delegate set of namespaces.
+
+This creates a serious challenge for the programs (like w3up client) that attempt to infer set of methods to expose based no delegated capabilities.
+
+We roughly two roles in the sysntem `space` and `account` and their capability set do not overlap, so we can somewhat infer which methods to present in case of `*` ability by subject DID, but that hinges upon the fact that we use `did:key` for space and `did:mailto` for accounts. This assumbtion may be rendered obsolete in the future.
```

Comment:
If I understand, we're infering methods based on the did type of the subject? I 100% agree let's not do this.